src/test/groovy/com/synacy/gradprogram/spock/exercise/DeliveryServiceSpec.groovy

```diff
@@ -0,0 +1,71 @@
+package com.synacy.gradprogram.spock.exercise
+
+import spock.lang.Specification
+import spock.lang.Unroll
+
+class DeliveryServiceSpec extends Specification {
+
+    DeliveryRequestRepository deliveryRequestRepository
+    Order order
+    DeliveryService deliveryService
+    DateUtils dateUtils
+
+    void setup() {
+        deliveryRequestRepository = Mock(DeliveryRequestRepository)
+        order = Mock(Order)
+        dateUtils = Mock(DateUtils)
+        deliveryService = new DeliveryService(dateUtils, deliveryRequestRepository)
+    }
+
+    def "CreateDelivery should save OrderID, DeliveryDate and Courier to deliveryRequestRepository"() {
```

Comment:
Although we are testing the "what" here, our description here is kinda detailing the "how". These two should go hand in hand for the same reason. Maybe something like this is better:
```
"createDelivery should create and save a delivery request for the given Order"
```

The details are abstracted from how it is doing things and is focused on what should happen. So for example if we need to refactor the code in the future to also add another field/property to the `DeliveryRequest` or if we are gonna save the `DeliveryRequest` using a different service (not the `deliveryRequestRepository`), the test method name/description will still be valid.